plugin/restore/mysql/mysql.go

```diff
@@ -144,6 +152,156 @@ func getPITROldDatabaseName(database string, suffixTs int64) string {
+				downloadedName[serverFile.Name] = struct{}{}
+			}
+		}
+	}
+
+	// download the binlog files not recorded in downloadedIndex
+	for _, serverFile := range binlogFilesOnServer {
+		if _, ok := downloadedName[serverFile.Name]; ok {
+			continue
+		}
+		if err := r.downloadBinlogFile(ctx, instance, saveDir, serverFile); err != nil {
+			return fmt.Errorf("cannot sync binlog %s, error: %w", serverFile.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// SyncLatestBinlog syncs the latest binlog between the instance and `saveDir`
+func (r *Restore) SyncLatestBinlog(ctx context.Context, instance *api.Instance, saveDir string) error {
+	latestBinlogFileOnServer, err := r.getLatestBinlogFileMeta(ctx)
+	if err != nil {
+		return err
+	}
+	return r.downloadBinlogFile(ctx, instance, saveDir, *latestBinlogFileOnServer)
+}
+
+// downloadBinlogFile syncs the binlog specified by `meta` between the instance and local.
+func (r *Restore) downloadBinlogFile(ctx context.Context, instance *api.Instance, saveDir string, binlog mysql.BinlogFile) error {
+	// for mysqlbinlog binary, --result-file must end with '/'
+	resultFileDir := strings.TrimRight(saveDir, "/") + "/"
+	// TODO(zp): support ssl?
+	cmd := exec.CommandContext(ctx, filepath.Join(mysqlbinlogBinPath, "bin", "mysqlbinlog"),
+		binlog.Name,
+		fmt.Sprintf("--read-from-remote-server --host=%s --port=%s --user=%s --password=%s", instance.Host, instance.Port, instance.Username, instance.Password),
+		"--raw",
+		fmt.Sprintf("--result-file=%s", resultFileDir),
+	)
+	cmd.Stdout = os.Stderr
+	cmd.Stderr = os.Stderr
+
+	resultFilePath := filepath.Join(resultFileDir, binlog.Name)
+	if err := cmd.Run(); err != nil {
+		_ = os.Remove(resultFilePath)
+		return fmt.Errorf("cannot run mysqlbinlog, error: %w", err)
+	}
+
+	fi, err := os.Stat(resultFilePath)
+	if err != nil {
+		_ = os.Remove(resultFilePath)
+		return fmt.Errorf("cannot stat %s, error: %w", resultFilePath, err)
+	}
+	if fi.Size() != binlog.Size {
+		_ = os.Remove(resultFilePath)
+		return fmt.Errorf("download file %s size not match", resultFilePath)
+	}
+
+	return nil
+}
+
+// getBinlogFilesMetaOnServer returns the metadata of binlogs
+func (r *Restore) getBinlogFilesMetaOnServer(ctx context.Context) ([]mysql.BinlogFile, error) {
+	db, err := r.driver.GetDbConnection(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := db.QueryContext(ctx, `SHOW BINARY LOGS;`)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var binlogFiles []mysql.BinlogFile
+	for rows.Next() {
+		var binlogFile mysql.BinlogFile
+		var unused interface{}
+		if err := rows.Scan(&binlogFile.Name, &binlogFile.Size, &unused /*Encrypted column*/); err != nil {
+			return nil, err
+		}
+		binlogFiles = append(binlogFiles, binlogFile)
+	}
+	return binlogFiles, nil
+}
+
+// showLatestBinlogFile returns the metadata of latest binlog
+func (r *Restore) getLatestBinlogFileMeta(ctx context.Context) (*mysql.BinlogFile, error) {
```

Comment:
We have getBinlogInfo() from plugin/db/mysql/mysql.go doing similar things. Can we reuse to combine them somehow?